app/models/miq_request.rb

```diff
@@ -508,7 +509,8 @@ def self.create_request(values, requester, auto_approve = false)
     values[:src_ids] = Array.wrap(values[:src_ids]) unless values[:src_ids].nil?
     request_type = values.delete(:__request_type__) || request_types.first
     initiator = values.delete(:__initiated_by__) || 'user'
-    request = create!(:options => values, :requester => requester, :request_type => request_type, :initiated_by => initiator)
+    parent_id = values[:request_options].nil? ? nil : values[:request_options].delete(:parent_id)
```

Comment:
Minor but you can use `&.` for safe navigation, like `parent_id = values[:request_options]&.delete(:parent_id)` and it will return `nil` automatically for you if `values[:request_options]` is `nil`